### PR TITLE
[minor] add subdivisions to holidays

### LINKED ISSUE #1574: add_country_holidays is missing subdivisions
**Prerequisites**

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Checked the Answered Questions on the Github Discussion board: https://github.com/ourownstory/neural_prophet/discussions
      If you have the same question but the Answer does not solve your issue, please continue the conversation there.
    * Checked that your issue isn't already filed: https://github.com/ourownstory/neural_prophet/issues
      If you have the same issue but there is a twist to your situation, please add an explanation there.
    * Considered whether your issue might need further discussing before being defined as a feature request:
      Please [post an idea or feedback](https://github.com/ourownstory/neural_prophet/discussions/categories/ideas-feedback)

**Is your feature request related to a problem? Please describe.**

add_country_holidays is lacking subregion support. I would like to use the holidays for the specific province of Canada I am modelling. Subregions are supported in Facebook Prophet and  python-holidays

**Describe the solution you'd like**

Add subregion support

**Describe alternatives you've considered**

Generating the holidays manually and using the event interface

### PR DESCRIPTION
**Pull request by MaiBe-ctrl**
## :microscope: Background

- closes #1574 

## :crystal_ball: Key changes

- Explain the main changes introduced by this pull request for the reviewer.

## :clipboard: Review Checklist
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, added docstrings and data types to function definitions.
- [x] I have added pytests to check whether my feature / fix works.

Please make sure to follow our best practices in the [Contributing guidelines](https://github.com/ourownstory/neural_prophet/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MaiBe-ctrl** [comment]
chnage the country parameter to accept a dict or str to list and omit the new added parameter!

**Review by ourownstory — CHANGES_REQUESTED** [review]
merge `get_country_holidays` into `get_holidays_from_country` and move to `hdays_utils`

File: neuralprophet/forecaster.py [hunk 1]
```diff
@@ -764,8 +764,8 @@ def add_country_holidays(
 
         Parameters
         ----------
-            country_name : str, list
-                name or list of names of the country
+            country_name : str, list, dict
+                name or list of names of the country or a dictionary where the key is the country name and the value is a list of subdivisions
```
Comment (ourownstory) [inline neuralprophet/forecaster.py:768]: remove list  (of subdivisions)

File: neuralprophet/configure.py [hunk 2]
```diff
@@ -500,18 +500,58 @@ class Event:
...
     upper_window: int
     mode: str = "additive"
     reg_lambda: Optional[float] = None
     holiday_names: set = field(init=False)
 
     def init_holidays(self, df=None):
-        self.holiday_names = utils.get_holidays_from_country(self.country, df)
+        self.holiday_names = self.get_holidays_from_country(self.country, df)
+
+    def get_holidays_from_country(self, country: Union[str, Iterable[str], dict], df=None):
```
Comment (ourownstory) [inline neuralprophet/configure.py:515]: `self` does not appear to be accessed in this method. I think we can safely move it to `hdays_utils.py`.

**Review by ourownstory — APPROVED** [review]
Looking good overall! I just suggest to move the `get_holidays_from_country` to `hday_utils.py`

**Comment by ourownstory** [comment]
@MaiBe-ctrl Looks like you missed one utility function that was in `time_dataset.py`. I moved it to `hday_utils.py`. It was also not supporting subdivisions, so I added support for that, too. There is a lot of simplification potential to reduce the code in  `hday_utils.py`, but we can leave that for the refactoring post-JIT.

**Review by ourownstory — APPROVED** [review]
Feel free to merge once tests pass.

**Pull request merged**